src/stpipe/pipeline.py

```diff
@@ -181,25 +181,23 @@ def get_config_from_reference(cls, dataset, disable=None):
         # Iterate over the steps in the pipeline
         with cls._datamodels_open(dataset, asn_n_members=1) as model:
             if isinstance(model, Sequence):
-                for contained_model in model:
-                    input_class = contained_model.__class__()
-                    metadata = input_class
-                    metadata.update(contained_model, only='PRIMARY')
+                crds_parameters = model._models[0].get_crds_parameters()
+                crds_observatory = model.crds_observatory
             else:
-                input_class = model.__class__()
-                metadata = input_class
-                metadata.update(model, only='PRIMARY')
+                crds_parameters = model.get_crds_parameters()
+                crds_observatory = model.crds_observatory
 
         for cal_step in cls.step_defs.keys():
             cal_step_class = cls.step_defs[cal_step]
-            refcfg['steps'][cal_step] = cal_step_class.get_config_from_reference(metadata)
+            refcfg['steps'][cal_step] = cal_step_class.get_config_from_reference(crds_parameters,
+                                                                                 crds_observatory=crds_observatory)
```

Comment:
This line throws an error for me - seems as though crds_observatory isn't an input to this method. Removing it works just fine, though.

Reply:
Sorry, this comment was intended for L193 specifically - I removed `crds_observatory=crds_observatory` from the call and it runs without error.

Reply:
It was failing because `Pipeline.get_config_from_reference` also needed the `crds_observatory` parameter which I added in the latest commit.